#
# Updated but not checked in:
#   (will commit)
#
#	modified: xcb-util/icccm/Makefile.am
#	modified: xcb-util/image/Makefile.am
#	modified: xcb-util/reply/Makefile.am
#	modified: xcb-util/wm/Makefile.am
#
diff --git a/xcb-util/icccm/Makefile.am b/xcb-util/icccm/Makefile.am
index 337e410..fe57cef 100644
--- a/xcb-util/icccm/Makefile.am
+++ b/xcb-util/icccm/Makefile.am
@@ -7,11 +7,17 @@ xcbinclude_HEADERS = xcb_icccm.h
 
 AM_CFLAGS = -Wall
 
+XCB_ATOM_CFLAGS = -I$(top_builddir)/atom
+XCB_ATOM_LIBS = $(top_builddir)/atom/libXCBAtom.la
+XCB_EVENT_CFLAGS = -I$(top_builddir)/event
+XCB_EVENT_LIBS = $(top_builddir)/event/libXCBEvent.la
+XCB_PROPERTY_CFLAGS = -I$(top_builddir)/property
+XCB_PROPERTY_LIBS = $(top_builddir)/property/libXCBProperty.la
+
 libXCBICCCM_la_SOURCES = icccm.c
-libXCBICCCM_la_CFLAGS = $(XCB_CFLAGS) -I../atom -I../event -I../property
-libXCBICCCM_la_LIBADD = $(XCB_LIBS) \
-	-L../atom -lXCBAtom \
-	-L../property -lXCBProperty
+libXCBICCCM_la_CFLAGS = $(XCB_CFLAGS) $(XCB_ATOM_CFLAGS) $(XCB_EVENT_CFLAGS) \
+	$(XCB_PROPERTY_CFLAGS)
+libXCBICCCM_la_LIBADD = $(XCB_LIBS) $(XCB_ATOM_LIBS) $(XCB_PROPERTY_LIBS)
 
 pkgconfig_DATA = xcb-icccm.pc
 
diff --git a/xcb-util/image/Makefile.am b/xcb-util/image/Makefile.am
index 8bae709..6e34430 100644
--- a/xcb-util/image/Makefile.am
+++ b/xcb-util/image/Makefile.am
@@ -7,9 +7,13 @@ xcbinclude_HEADERS = xcb_image.h
 
 AM_CFLAGS = -Wall
 
+XCB_AUX_LIBS = $(top_builddir)/convenient/libXCBAux.la
+XCB_AUX_CFLAGS = -I$(top_builddir)/convenient
+XCB_IMAGE_LIBS = libXCBImage.la
+
 libXCBImage_la_SOURCES = xcb_image.c
-libXCBImage_la_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) -I../convenient
-libXCBImage_la_LIBADD = $(XCB_LIBS) $(XCB_SHM_LIBS) -L../convenient -lXCBAux
+libXCBImage_la_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) $(XCB_AUX_CFLAGS)
+libXCBImage_la_LIBADD = $(XCB_LIBS) $(XCB_SHM_LIBS) $(XCB_AUX_LIBS)
 
 pkgconfig_DATA = xcb-image.pc
 
@@ -18,14 +22,15 @@ EXTRA_DIST=xcb-image.pc.in
 noinst_PROGRAMS = test_xcb_image test_xcb_image_shm test_formats
 
 test_xcb_image_SOURCES = test_xcb_image.c
-test_xcb_image_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) -I../convenient
-test_xcb_image_LDADD = $(XCB_LIBS) -L../convenient -lXCBAux -lXCBImage
+test_xcb_image_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) $(XCB_AUX_CFLAGS)
+test_xcb_image_LDADD = $(XCB_LIBS) -L../convenient $(XCB_AUX_LIBS) \
+	$(XCB_IMAGE_LIBS)
 
 test_xcb_image_shm_SOURCES = test_xcb_image_shm.c
-test_xcb_image_shm_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) -I../convenient
+test_xcb_image_shm_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) $(XCB_AUX_CFLAGS)
 test_xcb_image_shm_LDADD = $(XCB_LIBS) $(XCB_SHM_LIBS) \
-	-L../convenient -lXCBAux -lXCBImage
+	$(XCB_AUX_LIBS) $(XCB_IMAGE_LIBS)
 
 test_formats_SOURCES = test_formats.c
-test_formats_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) -I../convenient
-test_formats_LDADD = $(XCB_LIBS) -L../convenient -lXCBAux -lXCBImage
+test_formats_CFLAGS = $(XCB_CFLAGS) $(XCB_SHM_CFLAGS) $(XCB_AUX_CFLAGS)
+test_formats_LDADD = $(XCB_LIBS) $(XCB_AUX_LIBS) $(XCB_IMAGE_LIBS)
diff --git a/xcb-util/reply/Makefile.am b/xcb-util/reply/Makefile.am
index 8fc5afb..504c68d 100644
--- a/xcb-util/reply/Makefile.am
+++ b/xcb-util/reply/Makefile.am
@@ -7,6 +7,8 @@ xcbinclude_HEADERS = xcb_reply.h
 
 AM_CFLAGS = -Wall
 
+XCB_REPLY_LIBS = libXCBReply.la
+
 libXCBReply_la_SOURCES = reply.c
 libXCBReply_la_CFLAGS = $(XCB_CFLAGS)
 libXCBReply_la_LIBADD = $(XCB_LIBS) -lpthread
@@ -18,5 +20,5 @@ EXTRA_DIST=xcb-reply.pc.in
 noinst_PROGRAMS = test_reply
 
 test_reply_CFLAGS = $(XCB_CFLAGS)
-test_reply_LDADD = $(XCB_LIBS) -lXCBReply
+test_reply_LDADD = $(XCB_LIBS) $(XCB_REPLY_LIBS)
 test_reply_SOURCES = test_reply.c
diff --git a/xcb-util/wm/Makefile.am b/xcb-util/wm/Makefile.am
index 9b43a93..4bbbf6a 100644
--- a/xcb-util/wm/Makefile.am
+++ b/xcb-util/wm/Makefile.am
@@ -7,19 +7,31 @@ xcbinclude_HEADERS = xcb_wm.h
 
 AM_CFLAGS = -Wall
 
+XCB_AUX_CFLAGS = -I$(top_builddir)/convenient
+XCB_AUX_LIBS = $(top_builddir)/convenient/libXCBAux.la
+XCB_ATOM_CFLAGS = -I$(top_builddir)/atom
+XCB_ATOM_LIBS = $(top_builddir)/atom/libXCBAtom.la
+XCB_EVENT_CFLAGS = -I$(top_builddir)/event
+XCB_EVENT_LIBS = $(top_builddir)/event/libXCBEvent.la
+XCB_PROPERTY_CFLAGS = -I$(top_builddir)/property
+XCB_PROPERTY_LIBS = $(top_builddir)/property/libXCBProperty.la
+XCB_ICCCM_CFLAGS = -I$(top_builddir)/icccm
+XCB_ICCCM_LIBS = $(top_builddir)/icccm/libXCBICCCM.la
+XCB_XCBWM_LIBS = libXCBWM.la
+
 libXCBWM_la_SOURCES = manage.c table.c
 libXCBWM_la_CFLAGS = $(XCB_CFLAGS) \
-	-I../convenient \
-	-I../atom \
-	-I../event \
-	-I../property \
-	-I../icccm
+	$(XCB_AUX_CFLAGS) \
+	$(XCB_ATOM_CFLAGS) \
+	$(XCB_EVENT_CFLAGS) \
+	$(XCB_PROPERTY_CFLAGS) \
+	$(XCB_ICCCM_CFLAGS)
 libXCBWM_la_LIBADD = $(XCB_LIBS) \
-	-L../convenient -lXCBAux \
-	-L../atom -lXCBAtom \
-	-L../event -lXCBEvent \
-	-L../property -lXCBProperty \
-	-L../icccm -lXCBICCCM
+	$(XCB_AUX_LIBS) \
+	$(XCB_ATOM_LIBS) \
+	$(XCB_EVENT_LIBS) \
+	$(XCB_PROPERTY_LIBS) \
+	$(XCB_ICCCM_LIBS)
 
 pkgconfig_DATA = xcb-wm.pc
 
@@ -29,14 +41,16 @@ noinst_PROGRAMS = xcbwm-test
 
 xcbwm_test_SOURCES = xcbwm-test.c reply_formats.c reply_formats.h
 xcbwm_test_CFLAGS = $(XCB_CFLAGS) \
-	-I../convenient \
-	-I../atom \
-	-I../event \
-	-I../property \
-	-I../icccm
+	$(XCB_AUX_CFLAGS) \
+	$(XCB_ATOM_CFLAGS) \
+	$(XCB_EVENT_CFLAGS) \
+	$(XCB_PROPERTY_CFLAGS) \
+	$(XCB_ICCCM_CFLAGS)
 xcbwm_test_LDADD = $(XCB_LIBS) \
-	-L../convenient -lXCBAux \
-	-L../event -lXCBEvent \
-	-L../property -lXCBProperty \
-	-L../icccm -lXCBICCCM \
-	-lXCBWM -lpthread
+	$(XCB_AUX_LIBS) \
+	$(XCB_ATOM_LIBS) \
+	$(XCB_EVENT_LIBS) \
+	$(XCB_PROPERTY_LIBS) \
+	$(XCB_ICCCM_LIBS) \
+	$(XCB_XCBWM_LIBS) \
+	-lpthread
